tests/file_service.cpp

```diff
@@ -34,6 +34,33 @@ TEST(file_service, serve_file_404)
     ASSERT_EQ(404, res.status_code);
 }
 
+
+TEST(file_service, serve_file_malicious)
+{
+    luna::server server;
+    auto router = server.create_router("/");
+
+    std::string path {"../../foo/confidential.txt"};
+    path = router->sanitize_path(path);
+    // Invalid path
+    ASSERT_TRUE(path == "");
+
+    path = "foo/bar/../../../confidential.txt";
```

Comment:
For completeness, we should add tests around paths with leading `/` and trailing `/`s as well.

Reply:
_e.g._ `/foo/bar/../../../confidential.txt` and `/foo/bar/../../` (because Luna tries to load an `index.html` file if the specified path is a folder)

Reply:
Can there be multiple leading/trailing `/` ? eg:

`////foo/bar/../../../confidential.txt`
`/foo/bar/../..////`

I'm thinking of striping them off before resolving the path.

Reply:
Is this the expected behaviour?

`/foo/bar/../../../confidential.txt` should resolve to `""`
`/foo/bar/../../` should resolve to "index.html" since the requested resource is a folder

Reply:
Current implementation preserves leading and trailing '/' in the input path. Let me know if this is inline with your thinking (I've added some tests for those cases too)